src/adaptors/UBCFFSubsetAdaptor.cpp

```diff
@@ -1206,7 +1206,9 @@ UBGraphicsGroupContainerItem *UBCFFSubsetAdaptor::UBCFFSubsetReader::parseIwbGro
 
 
 
-    foreach (QString key, strokesGroupsContainer.keys().toSet())
+    const auto keys = strokesGroupsContainer.keys();
+    const auto keySet = QSet<QString>(keys.begin(), keys.end());
+    for (const QString &key : keySet)
     {
```

Comment:
Unfortunately, this leads to a compile error. And I think it is not necessary to create a `QSet`, you could as well iterate over the `QList`. My proposal would be:

```c++
    const auto keys = strokesGroupsContainer.keys();

    for (const QString &key : keys)
```

Reply:
It compiles correctly for me. Which Qt / gcc version are you compiling on?
Anyway, you are probably right about the set.

Reply:
Ok, I see. It needs Qt >= 5.14. The range constructor was added there. I'm compiling with 5.12 for some platforms. Could we find a solution which works for 5.12 to 6.2?